inspire_disambiguation/app.py

```diff
@@ -0,0 +1,143 @@
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Disambiguation extension."""
+import json
+import os
+from json import JSONDecodeError
+
+from . import config
+
+
+class BeardConfig(object):
+    """Simple config manager available from the whole app.
+
+    It's a simple singleton so it will initialize only once.
+    Default values are set in `set_default_values` method.
+    Configuration is loaded from `config.py` located in the same folder as this file.
+    Additionally all environment variables with a name that starts with
+        `DISAMBIGUATION_` prefix will be loaded and will overwrite current values.
+        `DISAMBIGUATION_` prefix will be removed from env values when loaded.
+
+    Example:
+        $ export DISAMBIGUATION_ES_HOSTNAME = "localhost:9201"
+        $ python
+        >>> from inspire_disambiguation import conf
+        >>> conf['ES_HOSTNAME']
+        "localhost:9201"
+
+    ETHNICITY_DATA_PATH: Location of ethnicity data required for EthnicityEstimator
+        training.
+
+    ETHNICITY_MODEL_PATH: Location of dumped ethnicity model.
+
+    DISTANCE_MODEL_PATH: Location of dumped distance model.
+
+    CLUSTERING_N_JOBS: Number of processes to use for clustering.
+
+    DISPLAY_PROGRESS: When True displays progress bar during pairing process.
+
+    LOG_LEVEL: Minimum log level which will be displaying messages.
+
+    REDIS_PHONETIC_BLOCK_KEY: Key in redis from which script will read signature blocks
+        to re-cluster.
+
+    REDIS_URL: full redis url.
+
+    REDIS_TIMEOUT: Redis timeout.
+
+    SAMPLED_PAIRS_SIZE: The number of signature pairs we sample during training.
+
+        Since INSPIRE has ~3M curated signatures it would take too much time
+        to train on all possible pairs, so we sample ~1M pairs in such a way
+        that they are representative of the known clusters structure.
+
+        Note:
+
+            It MUST be a multiple of 12 for the reason explained in
+            :mod:`inspirehep.modules.disambiguation.core.ml.sampling`.
+
+    ES_HOSTNAME: hostname and port of elasticsearch.
+
+    ES_TIMEOUT: Timeout set for redis connection.
+
+    ES_MAX_QUERY_SIZE: Maximum size for one page of results from ES.
+        Note:
+             By default ES allows to get 10000 results at one page at once.
+    """
+
+    config_data = None
+    instance_path = os.path.dirname(os.path.abspath(__file__))
+    KEY_PREFIX = "DISAMBIGUATION_"
```

Comment:
why we need key_prefix?

Reply:
key_prefix is only for values you read from env. And it's required so you won't load everything you exported in your shell